src/main/java/com/o19s/es/ltr/ranker/normalizer/FeatureNormalizingRanker.java

```diff
@@ -0,0 +1,72 @@
+package com.o19s.es.ltr.ranker.normalizer;
+
+import com.o19s.es.ltr.feature.FeatureNormalizerSet;
+import com.o19s.es.ltr.ranker.LtrRanker;
+import org.apache.lucene.util.Accountable;
+import org.apache.lucene.util.RamUsageEstimator;
+
+/**
+ * An LtrRanker that applies supplied normalizers to features prior
+ * to scoring
+ */
+public class FeatureNormalizingRanker implements LtrRanker, Accountable {
```

Comment:
Another strategy is to wrap feature queries with a NormalizingQueryWrapper that accepts the featureQuery and the normalizer. This would be built in the RankerQuery::createWeight when looping over the queries, the normalizers could be passed as a new field in LtrQueryContext.

The benefits I could see are:
- no need to loop again over the features
- we would not lose the explanation

not sure if it's worth effort redesign what you already wrote. If you are interested I think I may some code ready (or close to be ready) for this NormalizingQueryWrapper query class.

Reply:
I think that's a fine approach too... If it's small enough, want to give it a try on this branch?